learn/guides/ballerina-openapi-support.md

```diff
@@ -0,0 +1,312 @@
+---
+layout: ballerina-openapi-support-left-nav-pages-swanlake
+title: Ballerina OpenAPI Support 
+description: Check out how the Ballerina OpenAPI tooling makes it easy for users to start developing a service documented in the OpenAPI contract.
+keywords: ballerina, programming language, openapi, open api, restful api
+permalink: /learn/ballerina-openapi-support/
+active: openapi
+intro: OpenAPI Specification is a specification that creates a RESTFUL contract for APIs, detailing all of its resources and operations in a human and machine-readable format for easy development, discovery, and integration. Ballerina OpenAPI tooling will make it easy for users to start the development of a service documented in an OpenAPI contract in Ballerina by generating Ballerina service and client skeletons. It supports the user to take the code first API design approach by generating an OpenAPI contract for the given service implementation.
+redirect_from:
+  - /learn/how-to-use-openapi-tools/
+  - /learn/how-to-use-openapi-tools
+  - /learn/using-the-openapi-tools/
+  - /learn/using-the-openapi-tools
+  - /swan-lake/learn/using-the-openapi-tools/
+  - /swan-lake/learn/using-the-openapi-tools
+  - /learn/tooling-guide/cli-tools/openapi
+  - /learn/tooling-guide/cli-tools/openapi/
+  - /learn/ballerina-openapi-support/openapi
+  - /learn/cli-documentation/openapi/#using-the-capabilities-of-the-openapi-tools/
+  - /learn/cli-documentation/openapi/#using-the-capabilities-of-the-openapi-tools
+  - /learn/cli-documentation/openapi/#openapi-validator-compiler-plugin/
+  - /learn/cli-documentation/openapi/#openapi-validator-compiler-plugin
+  - /learn/cli-documentation/openapi/
+  - /learn/cli-documentation/openapi
+---
+
+## Using the Capabilities of the OpenAPI Tools
+
+The OpenAPI tools provide the following capabilities.
+ 
+ 1. Generate the Ballerina service stub or client stub for a given OpenAPI definition (use `openapi` command).
+ 2. Export the OpenAPI definition of a given Ballerina service (use `openapi` command).
+ 3. Validate service implementation of a given OpenAPI contract (annotate your Ballerina service with `openapi` annotation).
+    - The OpenAPI compiler plugin will allow you to validate a service implementation against an OpenAPI contract during
+  the compile time.This plugin ensures that the implementation of a service does not deviate from its OpenAPI contract.   
+ 
+ Prerequisite : Download and install the latest Ballerina version from [here](https://ballerina.io/downloads/).
```

Comment:
Do we need this `prerequisites`section in here?

Reply:
Earlier we didn't add `prerequisites` to learn a page in openapi, here add with the concern if some user comes to the page with a direct search ballerina openapi without going through the early step. I think we can remove this in here as well 🤔 
WDYT? shall I remove this :) ?